sim-lib/src/lib.rs

```diff
@@ -1037,14 +1038,15 @@ async fn produce_events<N: DestinationGenerator + ?Sized, A: PaymentGenerator +
             }
         }
 
-        let wait: Duration = if current_count == 0 {
-            let start = node_generator.payment_start();
-            if start != Duration::from_secs(0) {
-                log::debug!(
-                    "Using a start delay. The first payment for {source} will be at {:?}.",
-                    start
-                );
-            }
+        // On our first run, add a one-time start delay if set for our payment.
+        let opt_start = node_generator.payment_start();
```

Comment:
nit: This line doesn't really need to happen every loop. 

Calling `payment_start()` and assigning `opt_start` is only necessary on the first iteration. It might be worth moving that into the second condition so that `if current_count == 0` fails it will short circuit and not execute the `payment_start()` call.

```rust
 let wait = if current_count == 0 && node_generator.payment_start().is_some() {
      let start = node_generator.payment_start().unwrap(); // Safe due to is_some check above.
      log::debug!(
          "Using a start delay. The first payment for {source} will be at {:?}.",
          start
      );

      start
  } else {
...
```
Still not ideal to call `payment_start()` twice but I could not think of a way to get the value into the if block otherwise.

Reply:
> Still not ideal to call payment_start() twice but I could not think of a way to get the value into the if block otherwise.

Yeah, this was my original reasoning for creating `op_start` (+ learned that `if let` has more limited syntax than I thought). Agree that this is probably the best we can do - will add a note on the trait that it's expected to return the same value on every call.

Reply:
Actually just went for a more verbose set of branches, I think it's nicer than adding an external expectation on `payment_start`'s behavior (even if in practice it always is the same value) + added a comment explaining.

Reply:
Totally agree, that looks really good now. It is also easier to read and understand what is happening.